Libraries/Azure.psm1

```diff
@@ -1776,38 +1827,34 @@ Function Delete-ResourceGroup([string]$RGName, [switch]$KeepDisks, [bool]$UseExi
-                    $unlockedResources = @()
-                    # Get the lock for each resource and compute a list of "unlocked" resources
-                    foreach ($resource in $CurrentResources) {
-                        $resourceLock = Get-AzResourceLock -ResourceGroupName $RGName `
-                            -ResourceType $resource.ResourceType -ResourceName $resource.Name
-                        if (-not $resourceLock) {
-                            $unlockedResources += $resource
-                        }
+                while ($attempts -le 10) {
+                    # Get LISAv2 deployed resources and ResourceType is NOT 'AvailabilitySets'
+                    $currentResources = Get-AzResource -ResourceGroupName $RGName | `
+                        Where-Object {$_.ResourceType -inotmatch "availabilitySets" -and (($vmRoleNamePattern -and $_.Name -imatch "^($vmRoleNamePattern)") -or ($_.Name -imatch '^LISAv2-.+'))}
+                    # Get the lock for each resource and compute a list of "unlocked" resources, # Only try to delete the "unlocked" resources
+                    $unlockedResources = $currentResources | Where-Object {-not $(Get-AzResourceLock -ResourceGroupName $RGName -ResourceType $_.ResourceType -ResourceName $_.Name)}
+                    if (-not $unlockedResources) {
+                        break
                     }
-                    # Only try to delete the "unlocked" resources
-                    foreach ($resource in $unlockedResources) {
-                        Write-LogInfo "Removing resource $($resource.Name), type $($resource.ResourceType)"
-                        try {
-                            $current_resource = Get-AzResource -ResourceId $resource.ResourceId -ErrorAction Ignore
-                            if ($current_resource) {
-                                $null = Remove-AzResource -ResourceId $resource.ResourceId -Force -Verbose
-                            }
-                        } catch {
-                            Write-LogErr "Failed to delete resource $($resource.Name). We will try to remove it in next attempt."
-                        }
+                    else {
+                        RemoveAzResourcesByResourceType -ResourceType4iMatch 'Microsoft.Network/loadBalancers'
+                        RemoveAzResourcesByResourceType -ResourceType4iMatch 'Microsoft.Compute/virtualMachines'
+                        RemoveAzResourcesByResourceType -ResourceType4iMatch 'Microsoft.Network/networkInterfaces'
+                        RemoveAzResourcesByResourceType -ResourceType4iMatch 'Microsoft'
                     }
-                    $CurrentResources = $unlockedResources
                     $attempts++
                 }
-                Write-LogInfo "Resources in $RGName are deleted."
-                $retValue = $true
+                if ($attempts -le 10) {
+                    Write-LogInfo "Resources in $RGName are deleted."
```

Comment:
The line 1852 and 1855: Can you keep the sentence in past format? We keep all logs in past format. There are a few more in other files.

Reply:
updated, thanks!